apps/navstar_overlay/src/navstar_overlay_lashup_kv_listener.erl

```diff
@@ -0,0 +1,143 @@
+    {ok, listening, #data{ref = Ref}}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Initializes the server
+%%
+%% @spec callback_mode() -> handle_event_function | state_functions
+%% @end
+%%--------------------------------------------------------------------
+-spec(callback_mode() -> handle_event_function | state_functions).
+callback_mode() -> 
+    state_functions.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% This function is called by a gen_statem when it is about to
+%% terminate. It should be the opposite of Module:init/1 and do any
+%% necessary cleaning up. When it returns, the gen_statem terminates
+%% with Reason. The return value is ignored.
+%%
+%% @spec terminate(Reason, State, Data) -> void()
+%% @end
+%%--------------------------------------------------------------------
+-spec(terminate(Reason :: (normal | shutdown | {shutdown, term()} | term()),
+    State :: atom(), Data :: #data{}) -> term()).
+terminate(_Reason, _State, _Data) ->
+    ok.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Convert process state when code is changed
+%%
+%% @spec code_change(OldVsn, OldState, OldData, Extra) -> {state_functions, NewState, NewData}
+%% @end
+%%--------------------------------------------------------------------
+-spec(code_change(OldVsn :: term() | {down, term()}, OldState :: atom(),
+    OldData :: #data{}, Extra :: term()) ->
+    {ok, NewState :: atom(), NewData :: #data{}}).
+code_change(_OldVsn, OldState, OldData, _Extra) ->
+    {ok, OldState, OldData}.
+
+%%---------------------------------------------------------------------------------
+%% State transition
+%% 
+%%  +-----------+               +-------------+              +----------+
+%%  |           |               |             |              |          |
+%%  |           | lashup event  |             | lashup event |          | lashup event
+%%  | listening +-------------->| configuring +------------->| batching +------+
+%%  |           |               |             |              |          |      |
+%%  |           |               |             |<-------------+          |<-----+
+%%  |           |               |             |    timeout   |          |
+%%  +-----------+               +-------------+              +----------+
+%%
+%%---------------------------------------------------------------------------------
+
+listening(info, EventContent, Data) ->
+    lager:error("listening"),
+    {next_state, configuring, Data, {next_event, internal, EventContent}}.
+
+configuring(internal, {lashup_kv_events, Event = #{key := [navstar, overlay, Subnet], ref := Reference}},
+            Data = #data{ref = Ref}) when Ref == Reference, is_tuple(Subnet), tuple_size(Subnet) == 2 ->
+    lager:error("configuring-internal"),
+    Pid = navstar_overlay_configure:start(Event),
```

Comment:
Don't we want to wait for the configuration to complete the first time? So the system will get into a semi-consistent state?